contrib/golang/filters/http/source/common/dso/dso.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <dlfcn.h>
+
+#include <memory>
+#include <string>
+
+#include "source/common/common/logger.h"
+
+#include "absl/synchronization/mutex.h"
+#include "contrib/golang/filters/http/source/common/dso/libgolang.h"
+
+namespace Envoy {
+namespace Dso {
+
+class DsoInstance {
+public:
+  DsoInstance(const std::string dso_name);
+  ~DsoInstance();
+
+  GoUint64 envoyGoFilterNewHttpPluginConfig(GoUint64 p0, GoUint64 p1);
+  GoUint64 envoyGoFilterMergeHttpPluginConfig(GoUint64 p0, GoUint64 p1);
+
+  GoUint64 envoyGoFilterOnHttpHeader(httpRequest* p0, GoUint64 p1, GoUint64 p2, GoUint64 p3);
+  GoUint64 envoyGoFilterOnHttpData(httpRequest* p0, GoUint64 p1, GoUint64 p2, GoUint64 p3);
+
+  void envoyGoFilterOnHttpDestroy(httpRequest* p0, int p1);
+
+  bool loaded() { return loaded_; }
+
+private:
+  const std::string dso_name_;
+  void* handler_{nullptr};
+  bool loaded_{false};
+
+  GoUint64 (*envoy_go_filter_new_http_plugin_config_)(GoUint64 p0, GoUint64 p1) = {nullptr};
+  GoUint64 (*envoy_go_filter_merge_http_plugin_config_)(GoUint64 p0, GoUint64 p1) = {nullptr};
+
+  GoUint64 (*envoy_go_filter_on_http_header_)(httpRequest* p0, GoUint64 p1, GoUint64 p2,
+                                              GoUint64 p3) = {nullptr};
+  GoUint64 (*envoy_go_filter_on_http_data_)(httpRequest* p0, GoUint64 p1, GoUint64 p2,
+                                            GoUint64 p3) = {nullptr};
+
+  void (*envoy_go_filter_on_http_destroy_)(httpRequest* p0, GoUint64 p1) = {nullptr};
+};
+
+using DsoInstancePtr = std::shared_ptr<DsoInstance>;
+
+class DsoInstanceManager {
+public:
+  /**
+   * Load the go plugin dynamic library.
+   * @param dso_id is unique ID for dynamic library.
+   * @param dso_name used to specify the absolute path of the dynamic library.
+   * @return false if load are invalid. Otherwise, return true.
+   */
+  static bool load(std::string dso_id, std::string dso_name);
+
+  /**
+   * Get the go plugin dynamic library.
+   * @param dso_id is unique ID for dynamic library.
+   * @return nullptr if get failed. Otherwise, return the DSO instance.
+   */
+  static DsoInstancePtr getDsoInstanceByID(std::string dso_id);
+
+  /**
+   * Get the full amount of dynamic library information.
+   * @return all so ids separated by commas.
+   */
+  static std::string show();
```

Comment:
Is this used anywhere?

Reply:
> Is this used anywhere?

@mattklein123 This is a generic function for the DSO class, I will add some information about DSO runtime statistics in the next version. So the implementation of the function is kept for now.

Reply:
Please remove unused code. You can add it back in the next PRs. Same below. Thanks.

Reply:
@mattklein123  Done. Thanks for your comments.